force-app/main/default/classes/FormulaBuilderController.cls

```diff
@@ -0,0 +1,181 @@
+public with sharing class FormulaBuilderController {
```

Comment:
Let's decouple this a bit. Move the evaluation code to a class called FormulaEvaluator and have the FormulaBuilderController call it. There are going to be use cases where the FormulaEvaluator gets used even if there's no lwc or client sessions around (for example an automation or 'headless' flow that has no UI)

Reply:
@alexed1 Updated